crates/compiler/parse/src/ast.rs

```diff
@@ -549,6 +591,50 @@ impl<'a> Defs<'a> {
         let tag = EitherIndex::from_left(type_def_index);
         self.push_def_help(tag, region, spaces_before, spaces_after)
     }
+
+    // Find the first definition that is a Apply Suffixed
+    // We need the tag_index so we can use it to remove the value
+    // We need the value index to know if it is the first
+    pub fn search_suffixed_defs(&self) -> Option<(usize, usize)> {
+        for (tag_index, tag) in self.tags.iter().enumerate() {
```

Comment:
🎨 I generally prefer to avoid `continue` unless the alternative would be really complicated - so I'd write this loop body more like:

```rust
if let Err(value_index) = tag.split() {
    let index = value_index.index();

    if let ValueDef::Body(_, expr) = &self.value_defs[index] {
        if let Expr::Apply(sub_expr, _, _) = expr.value {
            if let Expr::Suffixed(_) = sub_expr.value {
                return Some((tag_index, index));
            }
        }
    }
}
```

You could also combine some of the other pattern matches, but `cargo fmt` kind of makes a mess of it:

```rust
if let Err(value_index) = tag.split() {
    let index = value_index.index();

    if let ValueDef::Body(
        _,
        Expr {
            value:
                Expr::Apply(
                    Expr {
                        value: Expr::Suffixed(_),
                        ..
                    },
                    _,
                    _,
                ),
            ..
        },
    ) = &self.value_defs[index]
    {
        return Some((tag_index, index));
    }
}
```

Reply:
That's a handy language feature. ❤️